src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier randomColor = new ColorSupplier();
```

Comment:
ColorSupplier object creates every time you call getRandomFigure/getDefaultFigure. Think of approach, when you create an object of ColorSupplier only once for this class. The same for Random